lib/vector-core/buffers/src/disk_v2/ledger.rs

```diff
@@ -364,12 +367,26 @@ impl Ledger {
     /// Tracks the statistics of a successful write.
     pub fn track_write(&self, record_size: u64) {
         self.state().increment_records(record_size);
+
+        // A 4GB record should not be possible to write, at all.
+        let record_size = record_size
+            .try_into()
+            .expect("record size didn't fit into usize; is this a 32-bit platform?");
```

Comment:
Notably we do provide builds for armv7 which is 32-bit. Would it make sense to flag out the disk buffers for that build? I believe we did previously do that for leveldb.

Alternatively, instead of `usize` could we just use a `u64` for `record_size`?

Reply:
I specifically use `usize` for _most_ byte-related things because most stdlib APIs use `usize` as the parameter for anything size-related: number of bytes written by `Write::write`, any `len()` method on containers, etc.

Might just be easier to flag it off for those platforms, yeah.

Reply:
I might be missing something, but it seems like we are converting from `u64` into `usize` here only to convert back to `u64` when publishing the metric via `counter!`. In that case, just using `u64`s all the way through would seemingly make sense to me.

Reply:
Addressed in d472e49.

I ended up not feature flagging the buffers, because thinking through where and when we used `usize` vs `u64`, my conclusion is that we have the right safeguards/fallible conversions in place in the sense of that we start from `usize` and go to `u64`, and then go back from `u64` to `usize`, so we'll fail loudly if that constraint is somehow violated, but as the design stands... it shouldn't actually ever be violated.